sat/cli/bootprep/validate.py

```diff
@@ -116,12 +116,17 @@ def validate_instance_schema_version(instance, schema_validator):
 
     # No default is needed here because the version property was added to the
     # bootprep_schema.yaml schema file simultaneously with this code.
-    current_version = schema_validator.schema['version']
-    strict_current_version = StrictVersion(current_version)
+    current_version = version.parse(schema_validator.schema['version'])
 
     try:
-        strict_instance_version = StrictVersion(instance_version)
-    except ValueError as err:
+        instance_version = version.parse(instance_version)
+
+        # LegacyVersion is deprecated (see https://github.com/pypa/packaging/issues/321),
+        # and sometimes invalid versions can parse as "Legacy". To avoid this,
+        # treat LegacyVersions as invalid.
+        if isinstance(instance_version, version.LegacyVersion):
+            raise version.InvalidVersion()
```

Comment:
nit: unused parens here (I think?)

Reply:
The parens are used since it's calling the constructor for `InvalidVerision()`.

Reply:
whoops, sounds good.

Reply:
The extra parens don't bother me (in fact I kinda like being explicit), but FWIW:

> The sole argument to [raise](https://docs.python.org/3/reference/simple_stmts.html#raise) indicates the exception to be raised. This must be either an exception instance or an exception class (a class that derives from [Exception](https://docs.python.org/3/library/exceptions.html#Exception)). If an exception class is passed, it will be implicitly instantiated by calling its constructor with no arguments:

https://docs.python.org/3/tutorial/errors.html#raising-exceptions

So if it derives from `Exception`, we don't need the parens.

Reply:
Huh, that's good to know!